File: code.leetcode/math/e258.各位相加.cpp
class Solution
{
public:
    int addDigits(int num)
    {
        // 遍历
        /*sum := 0不该在这*/
        while (num >= 10)
        {
            int sum = 0;
            if (num == 10)
            {
                num = 1;
            }
            while (num > 0)
            {
                sum += num % 10;
                num /= 10;
            }
            num = sum;
        }
        return num;
    }
};

File: code.leetcode/math/e258.各位相加.go
func addDigits(num int) int {
	//遍历
	/*sum := 0不该在这*/
	for num >= 10 {
		sum := 0
		if num == 10 {
			num = 1
		}
		for num > 0 {
			sum += num % 10
			num /= 10
		}
		num = sum
	}
	return num
}